src/Utility/FilteredIterator.php

```diff
@@ -30,11 +32,19 @@ final class FilteredIterator extends ArrayIterator {
 	 *
 	 * @param array $data
 	 * @param callable $callback Callback to be called on each value
+	 *
+	 * @throws \WpOrg\Requests\Exception\InvalidArgument When the passed $data argument is not iterable.
 	 */
 	public function __construct($data, $callback) {
+		if (InputValidator::is_iterable($data) === false) {
+			throw InvalidArgument::create(1, '$data', 'iterable', gettype($data));
+		}
+
 		parent::__construct($data);
 
-		$this->callback = $callback;
+		if (is_callable($callback)) {
```

Comment:
As mentioned in the comment, I'd stick with the check here, but remove the one in `current()` (which runs unnecessarily often).

Reply:
We can't remove the check in `current()` as `$callback` will be unset when the object is serialized and then unserialized (previous security fix).